lib/Optimizer/Dialect/CC/CCOps.cpp

```diff
@@ -970,11 +970,64 @@ struct FuseWithConstantArray
     return success();
   }
 };
+
+struct FoldQuantumStructMember
```

Comment:
I think we should just implement scalar replacement. "A rising tide lifts all boats."